src/libs/sys-util-rs/src/cgroup.rs

```diff
@@ -0,0 +1,690 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+// Copyright (c) 2019-2021 Ant Group
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::ops::Deref;
+use std::path::{Component, Path, PathBuf};
+use std::sync::Mutex;
+
+use cgroups::{Cgroup, CgroupPid, Controllers, Hierarchy, Subsystem};
+use lazy_static::lazy_static;
+use once_cell::sync::Lazy;
+
+use crate::sl;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error("Can not add tgid {0} to cgroup, {1:?}")]
+    AddTgid(u64, #[source] cgroups::error::Error),
+    #[error("failed to apply resources to cgroup: {0:?}")]
+    ApplyResource(#[source] cgroups::error::Error),
+    #[error("failed to delete cgroup after {0} retries")]
+    DeleteCgroup(u64),
+    #[error("Invalid cgroup path {0}")]
+    InvalidCgroupPath(String),
+}
+
+pub type Result<T> = std::result::Result<T, Error>;
+
+lazy_static! {
+    /// Disable cgroup v1 subsystems.
+    pub static ref DISABLED_HIERARCHIES: Mutex<Vec<String>> = Mutex::new(Vec::new());
+}
+
+/// Update the disabled cgroup subsystems.
+pub fn update_disabled_cgroup_list(hierarchies: &[String]) {
```

Comment:
`update_disabled_cgroup_list()` is not called on this PR so it would be good to explain when it would be called (and would it be called multiple times?)

Reply:
fixed